### DISCUSSION TITLE
Live previewing with Knap

### DISCUSSION THREAD

**Discussion by leanhdung1994** [WinEdt-Team/WinEdt-Support#53] [1.3 Ideas and features requests]
I am a fan of WYSIWYG feature. Recently, I have come across [Knap](https://github.com/frabjous/knap) which offers live rendering:

![latex-llpp](https://github.com/WinEdt-Team/WinEdt-Support/assets/28704090/2baf499d-2f99-4a30-8346-bccafb55cbe4)

On macOS, there are two editors that implement this feature.

1. [Texifier](https://www.texifier.com)

![live-typeset-recording-hello-world](https://github.com/WinEdt-Team/WinEdt-Support/assets/28704090/5ca2e0a0-c65a-41f5-aad1-2566024e4400)

2. [Compositor](https://compositorapp.com/)

![image](https://github.com/WinEdt-Team/WinEdt-Support/assets/28704090/d9164587-0626-4b84-81f0-d57c587c783d)

I hope @Alex-Simonic could integrate this feature of Knap into WinEdt 11. This would be very helpful for editing LaTeX documents with complicated equations. Thank you so much for your consideration!

**Comment by Alex-Simonic** [comment]
Thank you for all the links to the software that supposedly does wonderful things. I would appreciate seeing the most complicated equation that you ever had to typeset that would fit in the category: "This would be very helpful for editing LaTeX documents with complicated equations." All I saw in your promo video links was "Hello World". Send us a screen video of you typing that "very complicated" equation in all (or some) of the quoted apps. We may all learn from it. Or maybe not... And don't forget to measure the CPU consumption while doing so.

In any case, this is what I think: AI will eventually power ALL typesetting software (and the contents for better or worse) and WinEdt days are very much over as I am not willing to go there. In the long run, there may be no need for TeX, either. But AI will probably be the only "thing" that bothers "reading" this stuff as well.

In any case, you can use the software of your choice. There is no demand for WinEdt these days and adding your suggestions would not change that. I am past retirement age and at least for the next five months (or more) I won't touch WinEdt at all.

Let us know if you can assemble a team of capable programmers interested in working on a project like this [not called WinEdt] pro bono or whatever they think it is worth. I can share the source code with them as long as I don't have to do anything else (like changing their dippers:-) Or else follow the links and enjoy the available WYSIWYG TeX interfaces for what they are worth... No hard feelings on my side. Let the thousand flowers bloom...

**Reply by leanhdung1994** [reply]
Thank you for your reply.

1. I have just used WinEdt11 for a few days. I am very happy with its modern UI and fast compiling speed. It is superior to TeXworks and TeXstudio. There is no doubt that WinEdt11 is the most well-polished LaTeX editor on Windows.

2. Because I do not use MacOS, I could not give you my real use-case of Texifier and Compositor. Luckily, I am using [BaKoMa TeX](https://bakoma-tex.daniel-aldrich.ca/about/programs/overview) which is a true WYSIWYG LaTeX editor and is available on Windows. Unfortunately, the single developer of BaKoMa TeX passed away three years ago... So BaKoMa TeX will soon be unusable.

- Here is a recorded video of my real use-case: https://drive.google.com/file/d/1dg5aZc2PNy74JSaIy-XoT34UaC6Rb7wh/view?usp=sharing
- Here is the CPU consumption of above use-case: https://drive.google.com/file/d/1B0FxwCutG65h1iFaltizkyS4xLzQCV7M/view?usp=sharing

3. AI could now generate content about entertainment, but I think it will not be the case for LaTeX typesetting. When writing a research paper in LaTeX, I want to control every *single* detail of it. Especially, my research is about mathematics. I have no choice but to type by myself to ensure the correctness of the paper.

4. Finding such team of capable programmers (interested in working on such niche project about LaTeX) is like an impossible mission :( On the other hand, your WinEdt11 is already mature, so adding such a functionality about live preview to WinEdt11 is much more meaningful and feasible. The developer of [TeXpresso](https://github.com/let-def/texpresso) mentioned in [this comment](https://github.com/let-def/texpresso/issues/63#issuecomment-2148761091) that

> I am working on simplifying the build by reducing the number of dependencies, after that it should be possible to provide clearer instructions. I don't have much time so this will take a few months.

I feel that, after TeXpresso has a stable build, (with extensive experience of LateX and ultimate understanding of WinEdt11) you can integrate this great feature of TeXpresso into WinEdt11.

5. I know a lot of people are in love of this live preview. You can see them on [this Discord](https://discord.com/invite/kY4WSmuzx8) about BaKoMa TeX. I am willing to offer you a financial incentive (within my budget of a PhD student) to encourage you to implement this live preview in WinEdt11. MacOS has two editors (Texifier and Compositor) with my desired feature, but no editor on Windows does. It is so unfair for Windows users...

Thank you for your consideration.

**Reply by leanhdung1994** [reply]
On reflection, I have just realized that I have been too obsessed with this live preview and incremental compiling of BaKoMa TeX. It's time for me to let it go.

@Alex-Simonic I am deeply moved by your kindness that I have bought a license of WinEdt11...

![image](https://github.com/WinEdt-Team/WinEdt-Support/assets/28704090/8c7bd946-92e5-4a82-83a5-480e13d251a7)

All the best for you.
